File: Common/InMotionGIT.Common/Extensions/DataReaderExtensions.vb
#Region "using"

Imports System.Data.Common
Imports System.Runtime.CompilerServices

#End Region

Namespace Extensions

    ''' <summary>
    ''' Extension methods for the DbDataReader type
    ''' </summary>
    Public Module DataReaderExtensions

#Region "Numeric Extension"

        <Extension()>
        Public Function NumericValue(value As DbDataReader, name As String) As Decimal
            Dim ordinal As Integer = value.GetOrdinal(name)
            Dim result As Decimal = 0

            If Not IsNothing(value) AndAlso Not value.IsDBNull(ordinal) Then
                result = value.GetDecimal(ordinal)
            End If

            Return result
        End Function

        <Extension()>
        Public Function IntegerValue(value As DbDataReader, name As String) As Integer
            Dim ordinal As Integer = value.GetOrdinal(name)
            Dim result As Integer = 0

            If Not IsNothing(value) AndAlso Not value.IsDBNull(ordinal) Then
                result = value.GetInt32(ordinal)
            End If

            Return result
        End Function

#End Region

#Region "DateTime Extension"

        <Extension()>
        Public Function DateTimeValue(value As DbDataReader, name As String) As Date
            Dim ordinal As Integer = value.GetOrdinal(name)
            Dim result As Date = Date.MinValue

            If Not IsNothing(value) AndAlso Not value.IsDBNull(ordinal) Then
                result = value.GetDateTime(ordinal)
            End If

            Return result
        End Function

#End Region

#Region "Boolean Extension"

        <Extension()>
        Public Function BooleanValue(value As DbDataReader, name As String) As Boolean
            Dim ordinal As Integer = value.GetOrdinal(name)
            Dim result As Boolean = 0

            If Not IsNothing(value) AndAlso Not value.IsDBNull(ordinal) Then
                result = (value.GetInt32(ordinal) = 1)
            End If

            Return result
        End Function

        <Extension()>
        Public Function BooleanCharValue(value As DbDataReader, name As String) As Boolean
            Dim ordinal As Integer = value.GetOrdinal(name)
            Dim result As Boolean = 0

            If Not IsNothing(value) AndAlso Not value.IsDBNull(ordinal) Then
                result = (value.GetString(ordinal).Trim = "1")
            End If

            Return result
        End Function

#End Region

#Region "String Extension"

        <Extension()>
        Public Function StringValue(value As DbDataReader, name As String) As String
            Dim ordinal As Integer = value.GetOrdinal(name)
            Dim result As String = String.Empty

            If Not IsNothing(value) AndAlso Not value.IsDBNull(ordinal) Then
                result = value.GetString(ordinal)
                result = result.Trim
            End If

            Return result
        End Function

#End Region

    End Module

End Namespace

File: Common/InMotionGIT.Common.Core/Extensions/DataReaderExtensions.cs
using System;

#region using

using System.Data.Common;
using Microsoft.VisualBasic.CompilerServices;

#endregion using

namespace InMotionGIT.Common.Core.Extensions;

/// <summary>
/// Extension methods for the DbDataReader type
/// </summary>
public static class DataReaderExtensions
{
    #region Numeric Extension

    public static decimal NumericValue(this DbDataReader value, string name)
    {
        int ordinal = value.GetOrdinal(name);
        decimal result = 0m;

        if (!(value == null) && !value.IsDBNull(ordinal))
        {
            result = value.GetDecimal(ordinal);
        }

        return result;
    }

    public static int IntegerValue(this DbDataReader value, string name)
    {
        int ordinal = value.GetOrdinal(name);
        int result = 0;

        if (!(value == null) && !value.IsDBNull(ordinal))
        {
            result = value.GetInt32(ordinal);
        }

        return result;
    }

    #endregion Numeric Extension

    #region DateTime Extension

    public static DateTime DateTimeValue(this DbDataReader value, string name)
    {
        int ordinal = value.GetOrdinal(name);
        var result = DateTime.MinValue;

        if (!(value == null) && !value.IsDBNull(ordinal))
        {
            result = value.GetDateTime(ordinal);
        }

        return result;
    }

    #endregion DateTime Extension

    #region Boolean Extension

    public static bool BooleanValue(this DbDataReader value, string name)
    {
        int ordinal = value.GetOrdinal(name);
        bool result = Conversions.ToBoolean(0);

        if (!(value == null) && !value.IsDBNull(ordinal))
        {
            result = value.GetInt32(ordinal) == 1;
        }

        return result;
    }

    public static bool BooleanCharValue(this DbDataReader value, string name)
    {
        int ordinal = value.GetOrdinal(name);
        bool result = Conversions.ToBoolean(0);

        if (!(value == null) && !value.IsDBNull(ordinal))
        {
            result = value.GetString(ordinal).Trim() == "1";
        }

        return result;
    }

    #endregion Boolean Extension

    #region String Extension

    public static string StringValue(this DbDataReader value, string name)
    {
        int ordinal = value.GetOrdinal(name);
        string result = string.Empty;

        if (!(value == null) && !value.IsDBNull(ordinal))
        {
            result = value.GetString(ordinal);
            result = result.Trim();
        }

        return result;
    }

    #endregion String Extension
}